File: day-1-1.rb
file = File.open("./inputs/day-1.txt")
file_data = file.readlines.map(&:chomp)
file.close()
increases = 0
lastValue = 0
file_data.each do |number|
    if number.to_i > lastValue and lastValue > 0
        increases += 1 
    end 

    lastValue = number.to_i;
end

print increases

File: day-1-1.mjs
import { promises as fs } from "fs";

async function loadData() {
  return (await fs.readFile("inputs/day-1.txt")).toString();
}

const data = (await loadData()).split("\n");

const result = data.reduce((payload, next) => {
    next > payload.lastValue && payload.increases++;
    payload.lastValue = next;
    return payload;
}, {lastValue: 0, increases:0})

console.log(result.increases)

